group/assignment5/sridha2412.py

```diff
@@ -0,0 +1,71 @@
+from typing import List, Tuple
+
+def common_prefix(word1: str, word2: str) -> int:
+    """ A method to find the common prefix in two words and return the end position
+    Args:
+        word1: first word
+        word2: second word
+    Returns:
+        int: position end of common prefix
+    """
+
+    for i in range(0, min(len(word1), len(word2))):
+        if word1[i] != word2[i]:
+            break
+    return i
+
+
+def create_DAG(words: List[str]) -> dict:
+    """ A method to create a direct acyclic graph from a list of words
+    Args:
+        words: list of words e.g ['ART', 'RAT', 'CAT', 'CAR']
+    Returns:
+        dict: a DAG e.g. {'A': [R], 'R': ['C'], 'C', 'T': ['R']}
+    """
+
+    alphabets = set()
+    for word in words:
+        alphabets = alphabets.union(word)
+
+    graph = {alpha: {"set": set(), "visited": False} for alpha in alphabets}
+
+    for word1, word2 in zip(words, words[1:]):
+        cp = common_prefix(word1, word2)
+        if(word1[cp] != word2[cp]):
```

Comment:
Do we need to check it again?

Reply:
Oh yeah, you're right, I didn't realise that. Thanks!

Reply:
Oh wait, no. In case I have a word like 'CAR' and 'CART', it would return cp = 2 so I would need the check